### ISSUE TITLE
Offline Support

### ISSUE THREAD

**Issue by PostsDesert** [nkalupahana/baseline#389]
I'm occasionally in areas where a stable (or existent) internet connection isn't available. When this is the case, I'm unable to submit logs. It would be nice if the app offered an offline persistence option. Would this be possible by writing the queue of submissions to disk when offline and then resubmitting via the API after network connectivity is reestablished?

Current message when no network connection is available:
```
Something went wrong, please try again!
Error: Firebase: Error (auth/network-request-failed).
```

**Comment by nkalupahana** [comment]
I agree, and it seems like since the 1.6.0 release this is the most commonly requested feature. I have a full time job now, so development is a little slow :( but I'll try my best to get this out.

**Comment by aahmozart** [comment]
I agree, it would be very useful to have a queue of unsubmitted journals that can be submitted as soon as an internet connection is established. Many people including myself wanna go offline, but still want to maintain ability to write and record journals.
